test/e2e/util/proxy.go

```diff
@@ -0,0 +1,172 @@
+package util
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"strings"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "github.com/openshift/api/config/v1"
+	hyperv1 "github.com/openshift/hypershift/api/v1beta1"
+	"github.com/openshift/hypershift/cmd/cluster/core"
+	awsinfra "github.com/openshift/hypershift/cmd/infra/aws"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const CA_CONFIGMAP_KEY = "ca-bundle.crt"
+
+func EnsureProxyConfiguration(t *testing.T, ctx context.Context, client crclient.Client, opts core.CreateOptions, hostedCluster *hyperv1.HostedCluster) {
+	t.Run("EnsureProxyConfiguration", func(t *testing.T) {
+		g := NewWithT(t)
+
+		infraOptions := awsinfra.CreateInfraOptions{
+			Name:    opts.Name,
+			InfraID: opts.InfraID,
+		}
+
+		if hostedCluster.Spec.Platform.AWS.CloudProviderConfig.Subnet == nil || hostedCluster.Spec.Platform.AWS.CloudProviderConfig.Subnet.ID == nil {
+			t.Errorf("AWS.Subnet is not set")
+		}
+		subnetID := hostedCluster.Spec.Platform.AWS.CloudProviderConfig.Subnet.ID
+
+		if hostedCluster.Spec.Platform.AWS.CloudProviderConfig == nil {
+			t.Errorf("AWS.CloudProviderConfig is not set")
+		}
+		vpcID := hostedCluster.Spec.Platform.AWS.CloudProviderConfig.VPC
+
+		// generate self-signed CA
+		ca, key, err := generateCACert()
+		g.Expect(err).ToNot(HaveOccurred())
+
+		caAndKey := strings.Join([]string{key, ca}, "")
+		userData := []byte(fmt.Sprintf(squidProxyWithSSLScript, caAndKey))
+
+		ec2Client := Ec2Client(opts.AWSPlatform.AWSCredentialsFile, opts.AWSPlatform.Region)
+		proxyAddr, err := infraOptions.CreateProxyHost(ctx, opts.Log, ec2Client, *subnetID, vpcID, userData)
+		g.Expect(err).ToNot(HaveOccurred(), "failed to create proxy")
+
+		caConfigMap := corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "proxy-ca-bundle",
+				Namespace: hostedCluster.Namespace},
+			Data: map[string]string{
+				CA_CONFIGMAP_KEY: ca,
+			},
+		}
+
+		err = client.Create(ctx, &caConfigMap)
+		g.Expect(err).ToNot(HaveOccurred(), "failed to create CA configmap")
+
+		if hostedCluster.Spec.Configuration == nil {
+			hostedCluster.Spec.Configuration = &hyperv1.ClusterConfiguration{}
+		}
+		hostedCluster.Spec.Configuration.Proxy = &v1.ProxySpec{
+			HTTPProxy:  proxyAddr,
+			HTTPSProxy: proxyAddr,
+			TrustedCA: v1.ConfigMapNameReference{
+				Name: caConfigMap.Name,
+			},
+		}
+
+		err = client.Update(ctx, hostedCluster)
+		g.Expect(err).ToNot(HaveOccurred(), "failed to update hostedcluster proxy configuration")
+
+		// Wait until proxy configuration changes take effect
```

Comment:
where is this criteria coming from?

Reply:
Just an educated assumption from my testing. The proxy configuration needed sometime to propagate and (break things if not configured correctly). I don't have specific numbers on how long this takes

Reply:
When changing proxy we should see NodePoolUpdatingConfigConditionType in NodePools being true, so that could be the gate, i.e. don't move forward until  NodePoolUpdatingConfigConditionType=true, then wait for it to be false again and Nodes go healthy.

That said, once we put back the creation test we can postpone day 2 validation if you find it problematic to automate.